src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -556,13 +559,25 @@ public ImmutableArray<T> InsertRange(int index, IEnumerable<T> items)
             }
 
             T[] tmp = new T[self.Length + count];
-            Array.Copy(self.array, 0, tmp, 0, index);
-            int sequenceIndex = index;
-            foreach (var item in items)
+            
+            // CopyTo first in case bad interface impl
+            // tries to modify other parts of tmp
+            if (items is ICollection<T>)
             {
-                tmp[sequenceIndex++] = item;
+                ((ICollection<T>)items).CopyTo(tmp, index);
             }
-
+            else
+            {
+                // items is IReadOnlyCollection<T>
+                
```

Comment:
stylecop: there shouldn't be a blank line after a single-line comment.
(corefx folks can comment as to whether this is an enforced rule)